src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -98,4 +104,22 @@ public DarCollection updateDarCollection(Integer collectionId, User user) {
     darCollectionDAO.updateDarCollection(collectionId, user.getDacUserId(), new Date());
     return getByCollectionId(collectionId);
   }
+
+  //Writing this method here so it can be testable and reused for collection resources
+  //This method will be expanded to account for dataset binning once roles come into play
+  //For now it's functionality is simple, fetch the datasets and attach it to the collection
+  //DTO class is used to to it's dataUse attribute which will be needed for binning functionality (to be added soon)
+  public void addDatasetsToCollections(List<DarCollection> collections) {
+    for (DarCollection collection : collections) {
+      List<Integer> datasetIds = new ArrayList<Integer>();
+      for (DataAccessRequest dar : collection.getDars()) {
+        datasetIds.add(dar.getData().getDatasetIds().get(0));
+      }
+      List<Integer> distinctIds = datasetIds.stream().distinct().collect(Collectors.toList());
+      // Dataset call is being done seperatly to keep collection queries and collection mappers/reducers from getting messy
+      Set<DataSet> datasets = datasetDAO.findDatasetWithDataUseByIdList(distinctIds);
+      collection.setDatasets(datasets);
+    }
```

Comment:
This works, but will degrade performance pretty quickly due to the DAO call in a loop. One way to fix this is to adopt a batching strategy. 

1. Make a  map of collection id -> list dataset ids
2. Flatten all the dataset ids into a unique list
3. Query all the datasets in the list
4. Go back through the map and then find the datasets in the list of ids, and then add those to the collection.

The benefit is that in the loop case, you're making N database calls where N is the number of collections. In the batch case, you're making 1 database call regardless of the number of collections. All of the compute is done in processing the lists instead of separate queries to the db.